x-pack/plugins/observability/public/pages/slo_edit/components/slo_edit_form_indicator_section.tsx

```diff
@@ -7,47 +7,24 @@
 
 import { EuiFormRow, EuiPanel, EuiSelect, EuiSpacer } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
-import { CreateSLOInput } from '@kbn/slo-schema';
-import React, { useEffect } from 'react';
+import React from 'react';
 import { Controller, useFormContext } from 'react-hook-form';
-
 import { SLI_OPTIONS } from '../constants';
+import { useUnregisterFields } from '../hooks/use_unregister_fields';
+import { CreateSLOForm } from '../types';
 import { ApmAvailabilityIndicatorTypeForm } from './apm_availability/apm_availability_indicator_type_form';
 import { ApmLatencyIndicatorTypeForm } from './apm_latency/apm_latency_indicator_type_form';
 import { CustomKqlIndicatorTypeForm } from './custom_kql/custom_kql_indicator_type_form';
-import {
-  CustomMetricIndicatorTypeForm,
-  NEW_CUSTOM_METRIC,
-} from './custom_metric/custom_metric_type_form';
+import { CustomMetricIndicatorTypeForm } from './custom_metric/custom_metric_type_form';
 import { maxWidth } from './slo_edit_form';
 
 interface SloEditFormIndicatorSectionProps {
   isEditMode: boolean;
 }
 
 export function SloEditFormIndicatorSection({ isEditMode }: SloEditFormIndicatorSectionProps) {
-  const { control, watch, setValue } = useFormContext<CreateSLOInput>();
-
-  const indicator = watch('indicator.type');
-
-  useEffect(() => {
-    if (!isEditMode) {
-      if (indicator === 'sli.metric.custom') {
-        setValue('indicator.params.index', '');
-        setValue('indicator.params.timestampField', '');
-        setValue('indicator.params.good.equation', 'A');
-        setValue('indicator.params.good.metrics', [NEW_CUSTOM_METRIC]);
-        setValue('indicator.params.total.equation', 'A');
-        setValue('indicator.params.total.metrics', [NEW_CUSTOM_METRIC]);
-      }
-      if (indicator === 'sli.kql.custom') {
-        setValue('indicator.params.index', '');
-        setValue('indicator.params.timestampField', '');
-        setValue('indicator.params.good', '');
-        setValue('indicator.params.total', '');
-      }
-    }
-  }, [indicator, setValue, isEditMode]);
+  const { control, watch } = useFormContext<CreateSLOForm>();
+  useUnregisterFields({ isEditMode });
```

Comment:
💬  We reset the form on every SLI change we good default for all indicator types, so we don't need the following (now deleted) useEffect anymore